src/qaqc/01-organize_raw_data.Rmd

```diff
@@ -301,6 +303,70 @@ ggplot(temperature_archery, aes(x = DT_round, y = p1)) +
   theme_bw()
 ```
 
+```{r}
+# Determine each site and parameter in all_data 
+sites <- unique(all_data$site)
+params <- unique(all_data$parameter)
+# Constructing a df to iterate over each site-parameter combination
+combinations <- crossing(sites, params)
+
+# Function to summarize site-parameter combinations
+summarize_site_param <- function(site_arg, parameter_arg, api_data) {
+  
+  print(paste0('Trying site ', site_arg, " with parameter ", parameter_arg, "."))
+  
+  # filtering the data and generating results
+  summary <- tryCatch({ 
+    api_data %>%
+    filter(site == site_arg & parameter == parameter_arg) %>% 
+    select(-name) %>%
+    distinct() %>%
+    group_by(DT_round, site, parameter) %>%
```

Comment:
somewhere in here, I suggest adding in that filtering step discussed in previous comment.

Reply:
thought: for the purposes of {targets}, I'd keep the flagging for sonde out of water separate from this function.

The workflow would be (IMO): 1) collate data, 2) recode where out of water, 3) apply the summarization in this function... and then we'll probably have some global flags that are applicable to parameters across all sites, and then, finally, site/parameter specific funcitons.

Reply:
Agreed! I keep forgetting about {targets}. For {targets} we will want to keep each step in the QAQC process split up into their own functions, even regardless of whether they can be performed across all site-parameter combos or one at a time. I also agree 100% with B's overarching workflow steps he's laid out here.